src/main/scala/ai/eto/rikai/sql/model/Model.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 Rikai authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.eto.rikai.sql.model
+
+import org.apache.spark.sql.catalyst.expressions.Expression
```

Comment:
how tightly do we want to couple this with Spark?
e.g., would we ever want to integrate the ML SQL into say Presto or something? (assuming we're able to install and execute the models somehow).

Reply:
In a foresee future, I'd say it will be very tight (thus Spark SQL only), given how much integrations we need to do to provide the "Native feel" for any SQL engine.